### DISCUSSION TITLE
read tcp X.X.X.X I/o timeout

### DISCUSSION THREAD

**Discussion by WorstCodeWay** [bluenviron/mediamtx#3709] [Questions]
### Question

I'm sorry I'm new to stream or video streams, and I practice to implement video encoder with `FFmpeg` and `NVENC`.

When I create a new stream within ffmpeg c++ interface( just open a stream but don't put any packets into stream yet ), I do get log about rtsp is publishing to url ... But exactly after 10 seconds, I get timeout error, like:

```
2024/08/29 22:16:11 INF [RTSP] [session 61aef1d0] is publishing to path '1724560525331845/1724560527089504/OrignalStream', 1 track (H264)
2024/08/29 22:16:21 INF [RTSP] [conn 127.0.0.1:53628] closed: read tcp 127.0.0.1:8554->127.0.0.1:53628: i/o timeout
2024/08/29 22:16:21 INF [RTSP] [session 61aef1d0] destroyed: not in use
```

My questions are
1. What is the meaning of the I/o timeout here?
2. Is that designed behavior, I mean the session will be destroyed immediately when timeout is triggered?
3. Can I and how to set the timeout to "Never"? Means this session will never be destroyed, timeout is never triggered?

Thanks, guys.

**Comment by aler9** [comment]
answered in #3711
